imap_processing/codice/utils.py

```diff
@@ -88,7 +88,7 @@ def add_metadata_to_array(packet: space_packet_parser, metadata_arrays: dict) ->
     for key, value in packet.header.items():
         metadata_arrays.setdefault(key, []).append(value.raw_value)
 
-    for key, value in packet.data.items():
+    for key, value in packet.user_data.items():
```

Comment:
Similarly, are the types here correct?

Reply:
should we check with Gavin?

Reply:
It would be nice to get clarification from him

Reply:
Can't this be removed entirely because it is in `packet_file_to_datasets()` now? I thought codice was using that...

Reply:
yes. you are right Greg. I can delete this function. It's not used anymore. Sorry for including you to this PR by mistake.

Reply:
To your question Maxine, data type of both `header` and `user_data` are dict from [this](https://github.com/medley56/space_packet_parser/blob/main/space_packet_parser/packets.py#L126)

Reply:
This may be a Gavin question :)

Reply:
subclasses of built-in types.

An `IntParameter` is simply an `int` with a `raw_value` attribute attached to it. You should be able to use it exactly like an int, it just has an extra attribute to help you get at the raw_value if you really want it.